src/rules/noMisleadingArrayReverseRule.ts

```diff
@@ -38,34 +38,44 @@ export class Rule extends tslint.Rules.TypedRule {
     typescriptOnly: false,
   };
 
-  public static MESSAGE = "Move this array update to a separate call.";
+  public static MESSAGE = 'Move this array "reverse" operation to a separate statement.';
 
   public applyWithProgram(sourceFile: ts.SourceFile, program: ts.Program): tslint.RuleFailure[] {
     return this.applyWithWalker(new Walker(sourceFile, this.getOptions(), program));
   }
 }
 
 class Walker extends tslint.ProgramAwareRuleWalker {
-  public visitCallExpression(node: ts.CallExpression) {
+  public visitCallExpression(callExpression: ts.CallExpression) {
+    // first, take all call expressions: `foo()`
+
+    // then check that:
+    // * callee is a property access expression
+    // * left part of callee is array
+    // * the property name is "reverse": `foo.reverse()`
     if (
-      // a.b
-      this.isPropertyAccessExpression(node.expression) &&
-      // a.b or a.b.c
-      this.isIdentifierOrPropertyAccessExpression(node.expression.expression) &&
-      // a is array
-      this.isArray(node.expression.expression) &&
-      // only a.reverse
-      node.expression.name.text === "reverse" &&
-      this.isForbiddenOperation(node)
+      this.isPropertyAccessExpression(callExpression.expression) &&
+      this.isArray(callExpression.expression.expression) &&
+      callExpression.expression.name.text === "reverse"
     ) {
-      this.addFailureAtNode(node, Rule.MESSAGE);
+      const propertyAccessExpression = callExpression.expression.expression;
```

Comment:
@stas-vilchik not the best name. Property Access Expression is `a.b` but here you want only `a` (that's why this looks weird `this.isIdentifierOrPropertyAccessExpression(propertyAccessExpression)`). What about `arrayObjectNode`?